### ISSUE TITLE
Queue.playing and Queue.Connection.isPaused always seem to be wrong

### ISSUE THREAD

**Issue by royalty37** [Androz2091/discord-player#1487]
Whenever I attempt to write play and pause commands/subcommands in both a generic manner (such as how the guides/documentation recommend) or more similarly to others who have used discord-player, my player, queue and/or connection objects/instances never seem to behave correctly or hold the correct values. 

For example, when I call play, it will play a song, however if I call play while the song is already playing, it will nuke the currently playing song and play the new one... Logging out queue.playing reveals that it is always false, regardless of whether music is playing or not. Furthermore, I can simultaneously log out queue.playing (false) and queue.nowPlaying() (which will return the currently playing song) and get odd results...

Similarly, when I attempt to call pause, it will always say music isn't playing and therefore pause will say something like "Already paused...". I also tried to use queue.connection.paused but this always returns true even when music is playing.

This is my first attempt at creating a Discord bot and admittedly I'm quite stumped.

Any clues why? 

LINK TO MY REPO: https://github.com/royalty37/YO-DISCORD-BOT

Additionally, trackStart event is never emitted but I imagine that goes a little deeper (read through some of Discord-Player, Discord.JS but too much trouble).

**To Reproduce**
1. install and run project
2. Attempt to use /music play or /music pause

**Expected behavior**
They work correctly!!!!

 - Node Version: anywhere from 16.xxx to LTS
 - Discord Player Version: 5.3.1
 - Discord.js Version: 14.7.1

**Comment by Kief5555** [comment]
I'm having the same issue.

**Comment by Kief5555** [comment]
@Androz2091 Are you able to reproduce this?

**Comment by Kief5555** [comment]
Ah. I tried version 5.3.2. Seems like that does not have the issue. Cheers!

**Comment by XCOYoshi** [comment]
same with my implementation 5.4.0
repo: https://github.com/SJ-Plan-B/Plan-Bot

**Comment by Kief5555** [comment]
> same with my implementation in version 5.4.0
> 
> repo: https://github.com/SJ-Plan-B/Plan-Bot

Just use 5.3.2

**Comment by royalty37** [comment]
> > same with my implementation in version 5.4.0
> > repo: https://github.com/SJ-Plan-B/Plan-Bot
> 
> Just use 5.3.2

As mentioned in my original post, I'm having this issue with 5.3.1. Any ideas?

**Comment by Kief5555** [comment]
> > > same with my implementation in version 5.4.0
> 
> > > repo: https://github.com/SJ-Plan-B/Plan-Bot
> 
> > 
> 
> > Just use 5.3.2
> 
> 
> 
> As mentioned in my original post, I'm having this issue with 5.3.1. Any ideas?
> 
> 

Try to update .1 version? Lol. It works for me, returns `true` when isPlaying. 


to update you can do `npm i discord-player@5.3.2`

**Comment by skdhg** [comment]
I've been trying to reproduce this bug. In my case, this seems to be happening randomly.

**Comment by Kief5555** [comment]
> I've been trying to reproduce this bug. In my case, this seems to be happening randomly.

It happens every time for me. I'm not sure which version the standard `npm i discord-player` does but that was causing the issue, so I downgraded

**Comment by skdhg** [comment]
Sometimes `connection.paused` somehow seems to be `true` without setting it although it is initialized as `false`. I will change the `connection.paused` property to a getter returning `audioPlayer.state.status === AudioPlayerStatus.Paused` instead which should hopefully fix this issue as it returns what `audioPlayer` holds.

**Comment by skdhg** [comment]
I've opened a PR against this issue #1517

**Comment by skdhg** [comment]
After testing this for a while, I haven't faced this issue yet. This fix seems to be working but needs more tests. I could publish this version on `@dev` tag but the codebase is currently using extractors v4 dev version which may not be stable currently.

**Comment by XCOYoshi** [comment]
> 

still persistent in 5.3.2

**Comment by XCOYoshi** [comment]
> > same with my implementation in version 5.4.0
> > repo: https://github.com/SJ-Plan-B/Plan-Bot
> 
> Just use 5.3.2

still persistent in 5.3.2

**Comment by skdhg** [comment]
Fix for this https://github.com/Androz2091/discord-player/pull/1517 has been merged into dev branch. The fix will be out on stable once #1513 is merged into master.

**Comment by skdhg** [comment]
Hello, those who are having issues, could you try using `npm i discord-player@dev` and see if this problem still persists?

**Comment by XCOYoshi** [comment]
on it

**Comment by skdhg** [comment]
> > Hello, those who are having issues, could you try using `npm i discord-player@dev` and see if this problem still persists?
> 
> is it possible that the equalizer is not part of the dev branches because I am getting this error:
> 
> ```
> TypeError: import_equalizer.AudioFilter is not a constructor
>     at StreamDispatcher.createStream (I:\GIT\Plan-Bot\node_modules\discord-player\dist\index.js:1614:27)
>     at Queue.play (I:\GIT\Plan-Bot\node_modules\discord-player\dist\index.js:1401:38)
>     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
>     at async Object.execute (I:\GIT\Plan-Bot\commands\play.js:68:25)
>     at async Client.<anonymous> (I:\GIT\Plan-Bot\index.js:91:90)
> ```

🤔 does `npm i @discord-player/equalizer@latest` fix the problem?

**Comment by XCOYoshi** [comment]
that fixes it THX

**Comment by XCOYoshi** [comment]
the bug is gone to

**Comment by skdhg** [comment]
Starting with `discord-player@dev`, you don't need `onBeforeCreateStream` to use `play-dl`. Discord Player will automatically prefer `play-dl` over `ytdl-core` when it can detect `play-dl`.

**Comment by royalty37** [comment]
> possible that the equalizer is not part of the dev branches because I am getting this error:

I'm having a new issue after using the @dev version.

Now calling play will cause Queue to to throw a write EPIPE erorr...?

**Comment by Danrejk** [comment]
> Hello, those who are having issues, could you try using `npm i discord-player@dev` and see if this problem still persists?

I am unable to check.
When on the @dev version, the QueryType seems to be not working for me. It simply returns me: https://pastebin.com/8wWAjyBk
Instead of https://pastebin.com/KMRxa2BA, as it does in the 5.4.0 version

**Comment by skdhg** [comment]
> > Hello, those who are having issues, could you try using `npm i discord-player@dev` and see if this problem still persists?
> 
> I am unable to check. When on the @dev version, the QueryType seems to be not working for me. It simply returns me: https://pastebin.com/8wWAjyBk Instead of https://pastebin.com/KMRxa2BA, as it does in the 5.4.0 version

in dev version, `player.search` returns a `SearchResult` object instead of plain object. The search result object contains og `playlist` and `tracks` getters so it should be compatible with existing codebases

**Comment by skdhg** [comment]
However, the codebases that used direct values instead of `QueryType` will break as some changes were made to `QueryType` enum.

**Comment by skdhg** [comment]
> > Hello, those who are having issues, could you try using `npm i discord-player@dev` and see if this problem still persists?
> 
> I am unable to check. When on the @dev version, the QueryType seems to be not working for me. It simply returns me: https://pastebin.com/8wWAjyBk Instead of https://pastebin.com/KMRxa2BA, as it does in the 5.4.0 version

what I can see from your `SearchResult` object is you don't have extractors available. Unline before, discord-player no longer provides built-in extractors, everything has been moved to `@discord-player/extractor` package. Discord Player should install that automatically but if not then there is likely a problem of some dependencies.

Could you try running these?

`npm i play-dl`
`npm i @discord-player/extractor@latest`

**Comment by Danrejk** [comment]
Thanks for taking a look at it.
Both are up to date, and the SearchResult output is the same.

**Comment by Danrejk** [comment]
Honestly idk what was wrong, but I copied over the node_modules folder from a much older version of my bot, updated it to the @dev version and it works. So probobaly an error on my side. Sorry for the troubles.
